frontend/src/components/data-table/data-table.tsx

```diff
@@ -112,6 +121,15 @@ const DataTableInternal = <TData,>({
     onColumnFiltersChange: onFiltersChange,
     // selection
     onRowSelectionChange: onRowSelectionChange,
+    initialState: {
```

Comment:
regarding your question about shared state, 

i think you could do:

```
useEffect(() => {
    table.setState({columnPinning: {left: freezeColumnsLeft, freezeColumnsRight}})
}, [useDeepCompareMemoize([freezeColumnsLeft, freezeColumnsRight])]
```

im doing `useDeepCompareMemoize` since freezeColumnsLeft/freezeColumnsRight may change array instance, but the contents stay the same. its probably not that much more performant

Reply:
Looks like it was tunnel vision -- I did something similar and it worked. Thanks!